### PR TITLE
client-go: simplify deepcopy calls

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

File: staging/src/k8s.io/client-go/tools/cache/mutation_detector.go [hunk 1]
```diff
@@ -100,10 +99,7 @@ func (d *defaultCacheMutationDetector) AddObject(obj interface{}) {
 		return
 	}
 
-	copiedObj, err := scheme.Scheme.Copy(obj.(runtime.Object))
-	if err != nil {
-		return
-	}
+	copiedObj := obj.(runtime.Object).DeepCopyObject()
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/tools/cache/mutation_detector.go]: Combine it with line 98 to avoid a type assertion.

**Review by caesarxuchao — COMMENTED** [review]
A nit and then lgtm.

File: staging/src/k8s.io/client-go/tools/cache/mutation_detector.go (hunk 1 above)
Comment (sttts) [reply staging/src/k8s.io/client-go/tools/cache/mutation_detector.go]: done

**Pull request merged**